lib/methods/bulk-action.ts

```diff
@@ -0,0 +1,102 @@
+import delay from 'delay'
+import {
+  BulkAction,
+  BulkActionPayload,
+  BulkActionProps,
+  BulkActionStatus,
+} from '../entities/bulk-action'
+import { PlainClientAPI } from '../plain/common-types'
+
+const DEFAULT_MAX_RETRIES = 30
+const DEFAULT_INITIAL_DELAY_MS = 1000
+const DEFAULT_RETRY_INTERVAL_MS = 2000
+
+export class BulkActionProcessingError extends Error {
+  public bulkAction?: BulkAction
+
+  constructor(message: string, bulkAction?: BulkAction) {
+    super(message)
+    this.bulkAction = bulkAction
+    this.name = 'BulkActionProcessingError'
+  }
+}
+
+export type BulkActionProcessingOptions = {
+  /** The amount of times to retry. Defaults to 30 */
+  retryCount?: number
+  /** The interval between retries, in miliseconds (ms). Defaults to 2000 (2s) */
+  retryIntervalMs?: number
+  /**
+   * Initial delay in miliseconds when performing the first check.
+   * This is used to prevent short running bulkActions of waiting too long for a result.
+   * Defaults to 1000ms (1s)
+   * */
+  initialDelayMs?: number
+}
+
+/**
+ * @private
+ * @description Waits for a BulkAction to be completed and to be in one of the final states (failed or succeeded)
+ * @param {Function} fn - GET function that will be called every interval to fetch a bulkAction status
+ */
+export async function pollBulkActionStatus(
+  getBulkActionFunction: Function,
+  options?: BulkActionProcessingOptions
+): Promise<BulkActionProps> {
+  let retryCount = 0
+  let done = false
+  let action: BulkAction | undefined
+
+  const maxRetries = options?.retryCount || DEFAULT_MAX_RETRIES
+  const retryIntervalMs = options?.retryIntervalMs || DEFAULT_RETRY_INTERVAL_MS
+  const initialDelayMs = options?.initialDelayMs || DEFAULT_INITIAL_DELAY_MS
```

Comment:
I think the preferred approach is to use `??` nowadays instead of a coalesce `||` due to false-y value interpretation